android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/auth/AuthManager.java

```diff
@@ -0,0 +1,341 @@
+                    }
+                }).doOnError(new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        authManagerDelegateProtocol.setEmail(null);
+                        authManagerDelegateProtocol.setPassword(null);
+                    }
+                }).toCompletable();
+    }
+
+    /**
+     * @param email participant's email address
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable resendEmailVerification(@NonNull final String email) {
+        checkNotNull(email);
+
+        return RxUtils.toBodySingle(
+                authenticationApi.resendEmailVerification(
+                        new Email()
+                                .study(config.getStudyId())
+                                .email(email)))
+                .toCompletable();
+    }
+
+    /**
+     * On success, stores participant's email, password and session. If a
+     * NotAuthenticatedException is encountered, the participant's stored password and session
+     * are cleared.
+     *
+     * @param email    participant's email address
+     * @param password participant's password
+     * @return bridge response
+     */
+    @NonNull
+    public Single<UserSessionInfo> signIn(@NonNull final String email, @NonNull final String
+            password) {
+        checkNotNull(email);
+        checkNotNull(password);
+
+        SignIn signIn = new SignIn()
+                .study(config.getStudyId())
+                .email(email)
+                .password(password);
+
+        return RxUtils.toBodySingle(
+                authenticationApi.signIn(signIn))
+                .doOnSuccess(new Action1<UserSessionInfo>() {
+                    @Override
+                    public void call(UserSessionInfo userSessionInfo) {
+                        authManagerDelegateProtocol.setEmail(email);
+                        authManagerDelegateProtocol.setPassword(password);
+                        authManagerDelegateProtocol.setUserSessionInfo(userSessionInfo);
+                    }
+                }).doOnError(new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        if (throwable instanceof NotAuthenticatedException) {
+                            authManagerDelegateProtocol.setPassword(null);
+                        }
+                    }
+                });
+    }
+
+    /**
+     * On success, clears the participant's email, password and session from storage
+     *
+     * @return notifies of completion or error
+     */
+    @NonNull
+    public Completable signOut() {
+        return RxUtils.toBodySingle(authenticationApi.signOut())
+                .doOnSuccess(new Action1<Message>() {
+                    @Override
+                    public void call(Message message) {
+                        authManagerDelegateProtocol.setEmail(null);
```

Comment:
Yes, I'll add a TODO. I'm not sure yet how trigger ConsentManager to clean up the user's data and we probably want other Managers to clear out other data, like Tasks/Activities in the DB, maybe bridge data pending uploads (but maybe not)